frontend/src/metabase/browse/components/ModelsTable.tsx

```diff
@@ -126,9 +147,20 @@ export const ModelsTable = ({ models }: ModelsTableProps) => {
         </tr>
       </thead>
       <TBody>
-        {sortedModels.map((model: ModelResult) => (
-          <TBodyRow model={model} key={`${model.model}-${model.id}`} />
-        ))}
+        {error || isLoading ? (
+          <LoadingAndErrorWrapperTableRow>
+            <td colSpan={5}>
+              <DelayedLoadingAndErrorWrapper
+                error={error}
+                loading={!!isLoading}
+              />
+            </td>
+          </LoadingAndErrorWrapperTableRow>
+        ) : (
+          sortedModels.map((model: ModelResult) => (
+            <TBodyRow model={model} key={`${model.model}-${model.id}`} />
+          ))
+        )}
```

Comment:
putting the loading state inside the table is a nice UI touch, but I think it's unrelated to adding recents, so it would be best in a separate PR